src/actions/ratesActions.js

```diff
@@ -91,24 +114,33 @@ export const fetchSingleChainAssetRatesAction = (
   assetCode: string,
 ) => {
   return async (dispatch: Dispatch, getState: GetState) => {
-    const accountAssets = accountAssetsPerChainSelector(getState());
-    const chainAccountAssets = accountAssets[chain] ?? {};
+    const {
+      rates: { isFetching },
+      session: { data: { isOnline } },
+    } = getState();
 
-    const supportedAssets = supportedAssetsPerChainSelector(getState());
-    const chainSupportedAssets = supportedAssets[chain] ?? [];
+    if (isFetching || !isOnline) return;
+
+    dispatch(setIsFetchingRatesAction(true));
+
+    const accountAssetsPerChain = accountAssetsPerChainSelector(getState());
+    const chainAccountAssets = accountAssetsPerChain[chain] ?? {};
+
+    const supportedAssetsPerChain = supportedAssetsPerChainSelector(getState());
+    const chainSupportedAssets = supportedAssetsPerChain[chain] ?? [];
 
     const asset = getAssetData(getAssetsAsList(chainAccountAssets), chainSupportedAssets, assetCode);
     if (!asset) {
       reportErrorLog('fetchSingleChainAssetRatesAction failed: cannot find asset', { assetCode });
       return;
     }
 
-    const rates = await getExchangeRates({ [asset.symbol]: asset });
+    const rates = await getExchangeRates(chain, { [asset.symbol]: asset });
 
     if (isEmpty(rates)) return;
 
-    const { rates: { data: currentRates } } = getState();
-    const updatedRates = { ...currentRates, ...rates };
-    dispatch(setRatesAction(updatedRates));
+    await dispatch(setRatesAction(chain, rates));
```

Comment:
This looks like it would override record (for given chain) with various rates, with a new record with only one rate. Consider adding ```const updatedRates = { ...currentChainRates, ...rates };``` logic to ratesReducer.

Reply:
Actually I think this shouldn't be spread at all as we're setting, not updating, fixed.

Reply:
❗ 
This will overwrite all rates for given chain, with only one rate. Eg. store had rates for A, B, C but here we fetch rates for A and using current version of `setRates` will result in store no longer having rates for B, C.